### PR TITLE
Add depexts to caml-mode

### PR DESCRIPTION
**Pull request by Chris00**

### CODE REVIEW

File: packages/caml-mode/caml-mode.4.02/opam [hunk 1]
```diff
@@ -20,6 +20,11 @@ remove: [
   [make "uninstall-ocamltags" "BINDIR=%{bin}%"]
 ]
 depends: []
+depexts: [
+  [["debian"] ["emacs"]]
+  [["ubuntu"] ["emacs"]]
+  [["macosx"] ["emacs"]]
+]
```
Comment (avsm) [inline packages/caml-mode/caml-mode.4.02/opam]: thanks! If you could add `[["centos"]["emacs-nox"]]` then I think we have the majority covered.  You can get rid of the `macosx` line as its part of the base distribution there.

File: packages/caml-mode/caml-mode.4.02/opam [hunk 2]
```diff
@@ -20,6 +20,12 @@ remove: [
   [make "uninstall-ocamltags" "BINDIR=%{bin}%"]
 ]
 depends: []
+depexts: [
+  [["debian"] ["emacs-nox"]]
+  [["ubuntu"] ["emacs-nox"]]
+  [["macosx"] ["emacs"]]
```
Comment (avsm) [inline packages/caml-mode/caml-mode.4.02/opam]: This `macosx` line should definitely go -- the only interpretation that's valid right now is `[["osx"]["homebrew"]]` or `[["osx"]["macports"]]` and its not needed anyway due to Emacs being included by default in OSX.

**Comment by avsm** [comment]
Looks like Ubuntu doesn't have an `emacs-nox` package... perhaps just install `emacs` there?  Unsure about Debian.

**Comment by avsm** [comment]
https://travis-ci.org/ocaml/opam-repository/jobs/55564302 for the failure

**Comment by Chris00** [comment]
Debian does have `emacs-nox` so I though Ubuntu would do too.

**Comment by Chris00** [comment]
Let's try with `emacs` on Ubuntu... ;-)

**Comment by dbuenzli** [comment]
Le mardi, 24 mars 2015 à 14:53, Christophe Troestler a écrit :
> Let's try with emacs on Ubuntu... ;-)

FWIW the emacs packages are listed on this page http://packages.ubuntu.com/trusty/editors/  

It seems that the pattern is emacsXX-nox where XX is the emacs major version number.  

Best,

Daniel

**Comment by Chris00** [comment]
Thanks  @dbuenzli I would prefer not to have to update Emacs version number though so if `emacs` works, I would favor that.

**Comment by avsm** [comment]
Result! Thanks :)

**Pull request merged**